taskgraph/Task.py

```diff
@@ -969,9 +1017,10 @@ def _call(self):
                 execute='execute', fetch='one')
             try:
                 if database_result:
-                    result_target_path_stats = pickle.loads(
-                        database_result[0])
-                    LOGGER.debug('duplicate artifact db results: %s', result_target_path_stats)
+                    result_target_path_stats = pickle.loads(database_result[0])

```

Comment:
Right now I think of a taskgraph database as really a temporary cache. Nice if it works but not critical if it doesn't. But if this turned into a centralized database that had a long life span we should start thinking about things like this. I'm not so worried that we'd need to guard against a corrupt pickle specifically (would the database be corrupt)? But we should have a mechanism for rollbacks, resets, and consistency checks. We don't have a use case for that now, but if we did we should think about it the same way any kind of mission critical database should be designed. I don't know how to do that. I'd probably read a book.

Reply:
Yeah, I don't really know how to do that at well, and it feels like one of those things that's easy to conceive of and much harder to handle well.

The primary case I was worried about was the evolving standards for `pickle`.  For example, [python 3.4 introduced version 4, and 3.8 updated to version 5 of the pickle format](https://docs.python.org/3/library/pickle.html#data-stream-format), and I was worried that this update would break pickling.  At least for a simple test case of saving/loading a dict, I had no problems saving/loading between python 3.6 and 3.8 (which use different versions of the pickle spec).  I didn't try anything much more complicated (and there are some edge cases for `pickle` that can get hairy), but also we don't use `pickle` to save off super-complicated data structures and objects at this time.

So all this to say that what I was concerned about initially doesn't seem to be an issue, and yes, let's deal with how to handle this well when we start having a truly global state database we're maintaining in a cloud somewhere.